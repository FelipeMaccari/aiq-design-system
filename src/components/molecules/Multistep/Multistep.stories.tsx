import React from 'react'

import { Multistep } from './Multistep'

export default {
  component: Multistep,
  title: 'molecules/Multistep'
}

// eslint-disable-next-line react/prop-types
const Step: React.FC = ({ children }) => {
  return <h1>{children}</h1>
}

export const Basic: React.FC = () => {
  const steps = [
    { name: 'StepOne', component: <Step>One</Step> },
    { name: 'StepTwo', component: <Step>Two</Step> },
    { name: 'StepThree', component: <Step>Three</Step> },
    { name: 'StepFour', component: <Step>Four</Step> }
  ]
  return <Multistep stepActive={0} steps={steps} />
}
