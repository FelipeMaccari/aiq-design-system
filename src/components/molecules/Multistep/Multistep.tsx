import React from 'react'
import PropTypes from 'prop-types'

import styled from 'styled-components'

export interface Props {
  steps: {
    name: string
    component: any
  }[]
  stepActive: number
}

const MultistepStyled = styled.div``

export const Multistep: React.FC<Props> = () => {
  return <MultistepStyled />
}

Multistep.propTypes = {
  steps: PropTypes.array.isRequired,
  stepActive: PropTypes.number.isRequired
}
