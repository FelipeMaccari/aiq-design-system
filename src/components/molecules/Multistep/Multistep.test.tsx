import React from 'react'
import renderer from 'react-test-renderer'

import { Multistep } from './Multistep'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Multistep />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
