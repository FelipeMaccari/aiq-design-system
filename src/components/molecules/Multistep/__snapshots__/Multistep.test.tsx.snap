// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-AxirZ lVljt"
  width="100%"
>
  <div
    className="sc-AxirZ hfpubq"
    width="100%"
  >
    <div
      className="sc-AxirZ cCSTZZ"
      width="100%"
    >
      <button
        className="sc-AxhUy hlpokz"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-AxirZ dPXqAQ"
          height="24px"
          width="24px"
        >
          <span
            className="sc-AxjAm cKPSXY"
            color="white"
            cursor="pointer"
          >
            1
          </span>
        </div>
        <span
          className="sc-AxjAm jdpJgs"
          color="#eb2f96"
          cursor="pointer"
          fontSize="medium"
          fontWeight="medium"
        >
          StepOne
        </span>
      </button>
      <hr
        className="sc-AxhCb iSRrbn"
        height="1px"
        width="100%"
      />
    </div>
    <div
      className="sc-AxirZ cCSTZZ"
      width="100%"
    >
      <button
        className="sc-AxhUy hlpokz"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-AxirZ gIbMN"
          height="24px"
          width="24px"
        >
          <span
            className="sc-AxjAm cKPSXY"
            color="white"
            cursor="pointer"
          >
            2
          </span>
        </div>
        <span
          className="sc-AxjAm gXixKh"
          color="#dedede"
          cursor="pointer"
          fontSize="medium"
          fontWeight="medium"
        >
          StepTwo
        </span>
      </button>
      <hr
        className="sc-AxhCb iSRrbn"
        height="1px"
        width="100%"
      />
    </div>
    <div
      className="sc-AxirZ cCSTZZ"
      width="100%"
    >
      <button
        className="sc-AxhUy hlpokz"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-AxirZ gIbMN"
          height="24px"
          width="24px"
        >
          <span
            className="sc-AxjAm cKPSXY"
            color="white"
            cursor="pointer"
          >
            3
          </span>
        </div>
        <span
          className="sc-AxjAm gXixKh"
          color="#dedede"
          cursor="pointer"
          fontSize="medium"
          fontWeight="medium"
        >
          StepThree
        </span>
      </button>
      <hr
        className="sc-AxhCb iSRrbn"
        height="1px"
        width="100%"
      />
    </div>
    <div
      className="sc-AxirZ cCSTZZ"
      width="100%"
    >
      <button
        className="sc-AxhUy hlpokz"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-AxirZ gIbMN"
          height="24px"
          width="24px"
        >
          <span
            className="sc-AxjAm cKPSXY"
            color="white"
            cursor="pointer"
          >
            4
          </span>
        </div>
        <span
          className="sc-AxjAm gXixKh"
          color="#dedede"
          cursor="pointer"
          fontSize="medium"
          fontWeight="medium"
        >
          StepFour
        </span>
      </button>
    </div>
  </div>
  <div
    className="sc-AxiKw jTaWSa"
  >
    <h1>
      One
    </h1>
  </div>
</div>
`;
